modules/ximgproc/src/edge_drawing.cpp

```diff
@@ -251,6 +332,9 @@ void EdgeDrawingImpl::write(cv::FileStorage& fs) const
 EdgeDrawingImpl::EdgeDrawingImpl()
 {
     params = EdgeDrawing::Params();
+    nfa = new NFALUT(1, 1/2, 1, 1);
+    dH = new double[MAX_GRAD_VALUE];
+    grads = new int[MAX_GRAD_VALUE];
```

Comment:
> new

Where are these objects destroyed?

---

> dH, grads

Consider using `std::vector` for these fields.